analyzers/src/SonarAnalyzer.CSharp/Rules/DoNotCheckZeroSizeCollection.cs

```diff
@@ -52,13 +52,14 @@ private void AnalyzePropertyPatternClause(SyntaxNodeAnalysisContext c)
 
             foreach (var subPattern in propertyPatternClause.Subpatterns)
             {
-                if (subPattern.NameColon != null)
+                if (subPattern.ExpressionColon.SyntaxNode is NameColonSyntax nameColon)
```

Comment:
education: does this throw if any of the `ExpressionColon` or `SyntaxNode` is null?

Reply:
If my understanding is correct `ExpressionColon` should not be ever null while the SyntaxNode can be. The code will not break as we us the is pattern, however I added a code snippet to increase coverage. (Not buildable one)